proto/frequenz/api/common/v1/components.proto

```diff
@@ -75,46 +82,100 @@ enum ComponentCategory {
 
-  // Li-ion batteries.
-  BATTERY_TYPE_LI_ION = 1;
+  // The component is active.
+  COMPONENT_STATUS_ACTIVE = 1;
 
-  // Sodium-ion batteries
-  BATTERY_TYPE_NA_ION = 2;
+  // The component is inactive.
+  COMPONENT_STATUS_INACTIVE = 2;
 }
 
-// Enumerated inverter types.
-enum InverterType {
-  // Unspecified.
-  INVERTER_TYPE_UNSPECIFIED = 0;
+// Microgrid electrical component details.
+message Component {
+  // The component ID.
+  uint64 id = 1;
 
-  // Battery inverter.
-  INVERTER_TYPE_BATTERY = 1;
+  // Unique identifier of the parent microgrid_id.
+  uint64 microgrid_id = 2;
 
-  // Solar inverter.
-  INVERTER_TYPE_SOLAR = 2;
+  // The component name.
+  string name = 3;
 
-  // Hybrid inverter.
-  INVERTER_TYPE_HYBRID = 3;
-}
+  // The component category. E.g., Inverter, Battery, etc.
+  ComponentCategory category = 4;
+
+  // The metadata specific to the component category type.
+  ComponentCategoryMetadataVariant category_type = 5;
 
-// Enumerated EV charger types.
-enum EvChargerType {
-  // Default type.
-  EV_CHARGER_TYPE_UNSPECIFIED = 0;
+  // The component manufacturer.
+  string manufacturer = 6;
 
-  // The EV charging station supports AC charging only.
-  EV_CHARGER_TYPE_AC = 1;
+  // The model name of the component.
+  string model_name = 7;
 
-  // The EV charging station supports DC charging only.
-  EV_CHARGER_TYPE_DC = 2;
+  // The status of the component.
+  ComponentStatus status = 8;
 
-  // The EV charging station supports both AC and DC.
-  EV_CHARGER_TYPE_HYBRID = 3;
+  // The operational lifetime of the component.
+  frequenz.api.common.v1.lifetime.Lifetime operational_lifetime = 9;
+}
+
+// ComponentConnection describes a single electrical link between two components
+// within a microgrid, effectively representing the physical wiring as viewed
+// from the grid connection point, if one exists.
+//
+// !!! note "Physical Representation"
+//     This message is not about data flow but rather about the physical
+//     electrical connections between components. Therefore, the IDs for the
+//     source and destination components correspond to the actual setup within
+//     the microgrid.
+//
+// !!! note "Direction"
+//     The direction of the connection follows the flow of current away from the
+//     grid connection point, aligned with positive current according to the
+//     [Passive Sign Convention]
+//     (https://en.wikipedia.org/wiki/Passive_sign_convention).
+//
+// !!! info "Historical Data"
+//     The timestamps of when a connection was created and terminated allows for
+//     tracking the changes over time to a microgrid, providing insights into
+//     when and how the microgrid infrastructure has been modified.
+//
+message ComponentConnection {
+  // Unique identifier of the component where the connection originates. This is
+  // aligned with the direction of current flow away from the grid connection
+  // point, or in case of islands, away from the islanding point.
```

Comment:
I extended the original documentation slightly, in order to accommodate islands. If this looks good, I will add this to other places `ComponentConnection`'s documentation.

Reply:
Added this elsewhere. Adjusted the phrases slightly to go with the flow of the original documentation.